### ISSUE TITLE
failed to compile mini_racer_extension.cc on CentOS 6.10 without errors

### ISSUE THREAD

**Issue by yo-it-engineer** [discourse/mini_racer#147]
I had tried to install mini_racer on my CentOS 6.10 server but it has failed without any error output(or of what I seen).
It does output some warnings but I didn't see any error on the mini_racer side. Any instructions and help to find out what is causing the installation to fail is welcome.

```bash
$ gem install mini_racer -v '0.2.6' --source 'https://rubygems.org/'
Fetching: libv8-7.3.492.27.1-x86_64-linux.gem (100%)
Successfully installed libv8-7.3.492.27.1-x86_64-linux
Building native extensions.  This could take a while...
ERROR:  Error installing mini_racer:
	ERROR: Failed to build gem native extension.

    current directory: /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/mini_racer-0.2.6/ext/mini_racer_extension
/home/myserver/.rbenv/versions/2.4.5/bin/ruby -r ./siteconf20190708-14928-1x1ox1.rb extconf.rb
checking for -lpthread... yes
SETTING CXX
creating Makefile

current directory: /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/mini_racer-0.2.6/ext/mini_racer_extension
make "DESTDIR=" clean

current directory: /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/mini_racer-0.2.6/ext/mini_racer_extension
make "DESTDIR="
compiling mini_racer_extension.cc
cc1plus: warning: command line option '-Wimplicit-int' is valid for C/ObjC but not for C++ [enabled by default]
In file included from mini_racer_extension.cc:5:0:
/home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/v8.h:2681:5: error: looser throw specifier for 'virtual v8::String::ExternalStringResource::~ExternalStringResource()'
/home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/v8.h:2616:13: error:   overriding 'virtual v8::String::ExternalStringResourceBase::~ExternalStringResourceBase() noexcept (true)'
/home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/v8.h:2714:5: error: looser throw specifier for 'virtual v8::String::ExternalOneByteStringResource::~ExternalOneByteStringResource()'
/home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/v8.h:2616:13: error:   overriding 'virtual v8::String::ExternalStringResourceBase::~ExternalStringResourceBase() noexcept (true)'
In file included from /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/libplatform/libplatform.h:9:0,
                 from mini_racer_extension.cc:7:
/home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/libplatform/v8-tracing.h:240:3: error: looser throw specifier for 'virtual v8::platform::tracing::TracingController::~TracingController()'
In file included from /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/libplatform/v8-tracing.h:14:0,
                 from /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/libplatform/libplatform.h:9,
                 from mini_racer_extension.cc:7:
/home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/libv8-7.3.492.27.1-x86_64-linux/vendor/v8/include/v8-platform.h:124:11: error:   overriding 'virtual v8::TracingController::~TracingController() noexcept (true)'
cc1plus: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1plus: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1plus: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1plus: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
make: *** [mini_racer_extension.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/mini_racer-0.2.6 for inspection.
Results logged to /home/myserver/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0-static/mini_racer-0.2.6/gem_make.out
```

**Comment by ignisf** [comment]
From the README:

> Note using v8.h and compiling MiniRacer requires a C++11 standard compiler, more specifically clang 3.5 (or later) or gcc 4.8 (or later).

Does CentOS 6.10 offer such a compiler?

**Comment by yo-it-engineer** [comment]
> From the README:
> 
> > Note using v8.h and compiling MiniRacer requires a C++11 standard compiler, more specifically clang 3.5 (or later) or gcc 4.8 (or later).
> 
> Does CentOS 6.10 offer such a compiler?

Maybe not but I have compiled and installed gcc 7.3.0 on the server manually and tried to compiling MiniRacer again but it also shows the same errors.

Guess it doesn't use the newer version even if we have it manually compiled and installed. Or maybe there is a need to specify the version to use which I'm not aware of.

I ended up moving my deploy server to a new server running CentOS 7. Problem solved for my side.
